### PR TITLE
User/bsmithvt/get user permissions2

### PR DESCRIPTION
**Pull request by bsmithvt**

### CODE REVIEW

File: Fabric.Authorization.API/Modules/UsersModule.cs [hunk 1]
```diff
@@ -47,35 +57,56 @@ public class UsersModule : FabricModule<User>
...
+            var permissionResolutionResult = await new PermissionResolver(_roleService, _permissionService, Logger).Resolve(new PermissionResolutionRequest
+            {
+                SubjectId = SubjectId,
+                IdentityProvider = IdentityProvider,
+                Grain = userPermissionRequest.Grain,
+                SecurableItem = userPermissionRequest.SecurableItem,
+                UserGroups = await _userService.GetGroupsForUser(SubjectId, IdentityProvider)
+            });
+
+            return permissionResolutionResult.AllowedPermissions
```
Comment (michaelvidal24) [inline Fabric.Authorization.API/Modules/UsersModule.cs]: I think we want this response for the GetUserPermissions function.

File: Fabric.Authorization.API/Modules/UsersModule.cs [hunk 2]
```diff
@@ -47,35 +57,56 @@ public class UsersModule : FabricModule<User>
...
+                SecurableItem = userPermissionRequest.SecurableItem,
+                UserGroups = await GetGroupsForAuthenticatedUser(SubjectId, IdentityProvider)
+            });
 
             return new UserPermissionsApiModel
             {
                 RequestedGrain = userPermissionRequest.Grain,
                 RequestedSecurableItem = userPermissionRequest.SecurableItem,
-                Permissions = permissions
+                Permissions = permissionResolutionResult.AllowedPermissions.Except(permissionResolutionResult.DeniedPermissions).Select(p => p.Name)
```
Comment (michaelvidal24) [inline Fabric.Authorization.API/Modules/UsersModule.cs]: I think we want this response for the GetCurrentUserPermissions function.

File: Fabric.Authorization.API/Modules/UsersModule.cs [hunk 3]
```diff
@@ -47,35 +57,56 @@ public class UsersModule : FabricModule<User>
...
+            await CheckAccess(_clientService, userPermissionRequest.Grain, userPermissionRequest.SecurableItem,
+                AuthorizationReadClaim);
+
+            var permissionResolutionResult = await new PermissionResolver(_roleService, _permissionService, Logger).Resolve(new PermissionResolutionRequest
+            {
+                SubjectId = SubjectId,
+                IdentityProvider = IdentityProvider,
+                Grain = userPermissionRequest.Grain,
+                SecurableItem = userPermissionRequest.SecurableItem,
+                UserGroups = await _userService.GetGroupsForUser(SubjectId, IdentityProvider)
```
Comment (michaelvidal24) [inline Fabric.Authorization.API/Modules/UsersModule.cs]: For the current users permissions, we need to make sure we get the groups from the access token too.

File: Fabric.Authorization.Domain/Resolvers/Models/ResolvedPermission.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            return incomingPermission?.ToString().Equals(ToString(), StringComparison.OrdinalIgnoreCase) ?? false;
+        }
+
+        public override int GetHashCode()
+        {
+            return ToString().GetHashCode();
+        }
+    }
+
+    public static class ResolvedPermissionExtensions
```
[the remark this answers is not in the document]
Comment (bsmithvt) [reply Fabric.Authorization.Domain/Resolvers/Models/ResolvedPermission.cs]: Good point.

**Pull request closed without merging**